### PR TITLE
fixed bug that reset object on every iteration

### LINKED ISSUE #11: Support nested hashes
I've had no luck with getting the following to work:

```
<form>
  <input type='text' name='fruit[citrus][lemon]'>
  <input type='text' name='fruit[citrus][orange]'>
  <input type='text' name='fruit[citrus][lime]'>
</form>
```

__Desired output__
```
  fruit: {
    citrus: {
      lemon: $value,
      orange: $value,
      lime: $value
    }
  }
```

__Actual output__
```
  fruit: {
    citrus: {
      lime: $value
    }
  }
```

Before I start digging through the source to identify the problem, I thought I'd run it by here in case the cause is glaringly obvious to anybody.

### PR DESCRIPTION
**Pull request by izolate**
Fixes #11

### CODE REVIEW

**Comment by defunctzombie** [comment]
can we get a test for this

File: index.js [hunk 1]
```diff
@@ -174,7 +174,7 @@ function extract_from_brackets(result, key, value) {
         }
         else {
             // This is a nested key, set it properly for the next iteration
-            parent[child] = {};
+            parent[child] = parent[child] || {};
```
Comment (jxson) [inline index.js]: Good catch.

**Comment by jxson** [comment]
@izolate you could pretty easily modify the test "nested hashes with brackets" to cover this

**Comment by izolate** [comment]
I don't have credentials to run the tests. Can you help verify?

**Pull request merged**